status-im/status-react
# 📱➡️ 💳 Migrate on-device accounts to keycard

fixes #11431 

### Summary
Add an option to "Manage Keys" where a user can move the keys from a device to a Keycard. Helps bypass the cycle where a user needs to uninstall the app and recover from seed in-order to use a Keycard. Currently, the encrypted database is deleted, ie. chats will be lost. But this can be modified later to retain chats. 

### Review notes
This PR modifies some seed phrase recovery functions. The on-boarding process should be checked for mistakes.

### Testing notes


#### Platforms


- Android
- iOS

#### Areas that maybe impacted

##### Functional

- account recovery
- new account

### Steps to test
- On the multiaccounts list screen, click on a multiaccount
- Using the "Access existing keys" button, goto "Manage keys and storage"
- Follow the UI flow (Enter seed-> Select Keycard -> Confirm -> Acknowledge risk and confirm again)
- The Kecard onboarding flow should begin
- Follow the keycard flow and setup your account

In the background, the multiaccount is deleted from the device and moved to Keycard
